compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -74,6 +74,9 @@ impl<'tcx> Const<'tcx> {
 
         let ty = tcx.type_of(def.def_id_for_type_of());
```

Comment:
We should figure out why this returns an unnormalized type, we should not be normalizing here, as we don't have enough information available to properly normalize (e.g. we have no param env available).

Reply:
`type_of` always returns an unnormalized type, we can normalize the type in the param env of te environment of the constant i think